src/ui/components/SpecialDistribution.js

```diff
@@ -1,38 +1,73 @@
 // @flow
-
 import React, {useState} from "react";
-import {div, fromInteger, fromFloatString, ZERO} from "../../ledger/grain";
-import {type IdentityId} from "../../ledger/identity";
+import {Button, Container, TextField} from "@material-ui/core";
+import {Alert} from "@material-ui/lab";
+import {makeStyles} from "@material-ui/core/styles";
+import {div, fromFloatString, lt, ZERO} from "../../ledger/grain";
 import {type Account} from "../../ledger/ledger";
 import AccountDropdown from "./AccountSelector";
 import {computeAllocation} from "../../ledger/grainAllocation";
 import * as uuid from "../../util/uuid";
 import type {TimestampMs} from "../../util/timestamp";
 import {useLedger} from "../utils/LedgerContext";
 
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "80%",
+    minWidth: "1100px",
+    margin: "0 auto",
+    padding: "0 5em 5em",
+  },
+  arrowBody: {
+    color: theme.palette.text.primary,
+    flex: 1,
+    background: theme.palette.background.paper,
+    padding: "5px 20px",
+    display: "flex",
+    alignItems: "center",
+  },
+  triangle: {
+    width: 0,
+    height: 0,
+    background: theme.palette.background,
+    borderTop: "30px solid transparent",
+    borderBottom: "30px solid transparent",
+    borderLeft: `30px solid ${theme.palette.background.paper}`,
+  },
+  dropdownWrapper: {
+    border: "1px solid #9EADBA",
+    borderRadius: "50px",
+  },
+  centerRow: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  element: {flex: 1, margin: "20px"},
+  arrowInput: {width: "40%", display: "inline-block"},
+  headerText: {color: theme.palette.text.primary},
+}));
+
 export const SpecialDistribution = () => {
   const {ledger, updateLedger} = useLedger();
 
+  const classes = useStyles();
   const [credTimestamp, setCredTimestamp] = useState<TimestampMs>(+Date.now());
-  const [recipient, setRecipient] = useState<IdentityId | null>(null);
-  const [amount, setAmount] = useState<string>("0");
+  const [recipient, setRecipient] = useState<Account | null>(null);
+  const [amount, setAmount] = useState<string>("");
   const [memo, setMemo] = useState<string>("");
+  const [successMessage, setSuccessMessage] = useState<string>("");
 
-  const setRecipientFromAccount = (acct: Account | null) => {
-    setRecipient(acct ? acct.identity.id : null);
-  };
-
-  const submitDistribution = (e) => {
-    e.preventDefault();
-    if (recipient != null && amount !== ZERO) {
+  const submitDistribution = () => {
+    if (recipient && amount && amount !== ZERO) {
       const policy = {
         policyType: "SPECIAL",
         budget: fromFloatString(amount),
         memo,
-        recipient,
+        recipient: recipient.identity.id,
```

Comment:
The name of this field should be `recipientId` instead of `recipient` since its not a recipient, its just the id